src/main/java/rx/internal/operators/OperatorDoOnSubscribe.java

```diff
@@ -39,19 +40,6 @@ public OperatorDoOnSubscribe(Action0 subscribe) {
         subscribe.call();
         // Pass through since this operator is for notification only, there is
         // no change to the stream whatsoever.
-        return new Subscriber<T>(child) {
-            @Override
-            public void onNext(T t) {
-                child.onNext(t);
-            }
-            @Override
-            public void onError(Throwable e) {
-                child.onError(e);
-            }
-            @Override
-            public void onCompleted() {
-                child.onCompleted();
-            }
-        };
+        return Subscribers.wrap(child);
```

Comment:
This one is less obvious as to why it needs to be wrapped. The duplicate `onStart` problem seems like something we should fix inside `Subscriber` rather than needing to object allocate and wrap.